### DISCUSSION TITLE
Set static input size for rtmdet conversion to onnx for onnxruntime backend

### DISCUSSION THREAD

**Discussion by mcusgithub** [open-mmlab/mmdeploy#2096] [Q&A]
I'm able to convert rtmdet to onnx as described in the [documentation](https://github.com/open-mmlab/mmdeploy/blob/main/docs/en/04-supported-codebases/mmdet.md#convert-model) for the onnxruntime backend with:

1. dynamic input shape:
python3 tools/deploy.py configs/mmdet/detection/detection_onnxruntime_dynamic.py /mmdetection/configs/rtmdet/rtmdet_s_8xb32-300e_coco.py /mmdetection/configs/rtmdet/rtmdet_s_8xb32-300e_coco_20220905_161602-387a891e.pth demo/resources/det.jpg --work-dir ./work_dir --device cpu

2. static input shape:
python3 tools/deploy.py configs/mmdet/detection/detection_onnxruntime_static.py /mmdetection/configs/rtmdet/rtmdet_s_8xb32-300e_coco.py /mmdetection/configs/rtmdet/rtmdet_s_8xb32-300e_coco_20220905_161602-387a891e.pth demo/resources/det.jpg --work-dir ./work_dir --device cpu

I can't however figure out how to specify a static input shape by modifying the mmdet .py configuration file
- I know it can be done for the tensorrt backend using configs/mmdet/detection/detection_tensorrt_static-**640x640**.py for instance
- I know it can be done in mmyolo with projects/easydeploy/tools/export.py ... **--img-size 320 640 --batch 1** ...

I tried to change the default static configuration in various ways, like for instance: 
```
onnx_config = dict(input_shape=[320, 640])

backend_config = dict(
    model_inputs=[
        dict(
            input_shapes=dict(
                input=dict(
                    min_shape=[1, 3, 320, 640],
                    opt_shape=[1, 3, 320, 640],
                    max_shape=[1, 3, 320, 640])))
    ])
```
but the parameters are always ignored.

Can someone help me figuring it out?
Thanks a lot.

**Comment by irexyc** [comment]
@mcusgithub 

> I tried to change the default static configuration in various ways, like for instance:

For onnxruntime backend, you should use deploy configs for tensorrt.

When you use `deploy.py` and want a custom static input shape. For tensorrt, you can use dynamic deploy config but set min/opt/max shape to the same value. For onnxruntime, as mmdeploy will use `test_pipeline` to create input for model convertion (https://github.com/open-mmlab/mmdeploy/blob/main/mmdeploy/codebase/mmdet/deploy/object_detection.py#L166-L223), the key it to create a input with custom size. Take `rtmdet_s_8xb32-300e_coco.py` for example, as it inherit `test_pipeline` in `rtmdet_l_8xb32-300e_coco.py`, you can edit the these [two lines](https://github.com/open-mmlab/mmdetection/blob/main/configs/rtmdet/rtmdet_l_8xb32-300e_coco.py#L103-L104) to create the input shape you want.
